felix/bpf/proxy/syncer.go

```diff
@@ -194,9 +200,15 @@ func uniqueIPs(ips []net.IP) []net.IP {
 
 // NewSyncer returns a new Syncer
 func NewSyncer(nodePortIPs []net.IP,
-	svcsmap *cachingmap.CachingMap[nat.FrontendKey, nat.FrontendValue],
-	epsmap *cachingmap.CachingMap[nat.BackendKey, nat.BackendValue],
-	affmap maps.Map, rt Routes) (*Syncer, error) {
+	svcsmap *cachingmap.CachingMap[nat.FrontendKeyInterface, nat.FrontendValue],
+	epsmap *cachingmap.CachingMap[nat.BackendKey, nat.BackendValueInterface],
+	affmap maps.Map, rt Routes,
+	newFEK func(addr net.IP, port uint16, protocol uint8) nat.FrontendKeyInterface,
```

Comment:
Think the API would be cleaner if you passed in a `KVFactory` interface with these functions as methods.  Then you'd just need two static instances of that interface and all the constructor calls would be a lot more concise.

Reply:
I only pass the family and the rest is internal to the syncer.